packages/components/resource-pages/list-page.tsx

```diff
@@ -0,0 +1,264 @@
+    useK8sWatchResource<BucketClassKind[]>(bucketClassResource);
+  const bucketPolicyMap = React.useMemo(
+    () =>
+      bucketPoliciesLoaded && !bucketPoliciesError
+        ? bucketPolicies?.reduce((bcMap, bc) => {
+            const namespacePolicy = bc?.spec?.namespacePolicy;
+            if (namespacePolicy?.type === BucketClassType.SINGLE) {
+              const ns = namespacePolicy?.single?.resource;
+              bcMap[ns] = bcMap[ns] + 1 || 1;
+            } else if (namespacePolicy?.type === BucketClassType.MULTI) {
+              const dataResourceSet = new Set<string>();
+              dataResourceSet.add(namespacePolicy?.multi?.writeResource);
+              namespacePolicy?.multi?.readResources.forEach((r) =>
+                dataResourceSet.add(r)
+              );
+              dataResourceSet.forEach((ns) => (bcMap[ns] = bcMap[ns] + 1 || 1));
+            }
+            return bcMap;
+          }, {})
+        : {},
+    [bucketPolicies, bucketPoliciesLoaded, bucketPoliciesError]
+  );
+
+  const [dataResources, dataResourcesLoaded, dataResourcesError] =
+    useK8sWatchResource<K8sResourceCommon[]>(nameSpaceStoreResource);
+  const [data, filteredData, onFilterChange] = useListPageFilter(dataResources);
+  const loaded = dataResourcesLoaded && bucketPoliciesLoaded;
+  const loadError = dataResourcesError || bucketPoliciesError;
+  const createLink = `/mcgms/resource/${referenceForModel(
+    NooBaaNamespaceStoreModel
+  )}/create/~new`;
+
+  return (
+    <>
+      <Modal {...modalProps} />
+      <div>
+        <div className="filter-data-resource__search-box">
+          <ListPageFilter
+            data={data}
+            loaded={loaded}
+            onFilterChange={onFilterChange}
+            hideColumnManagement={true}
+            nameFilterPlaceholder={t("Filter by policy name")}
+            labelFilterPlaceholder={t("Filter by policy label")}
+          />
+        </div>
+        <div className="create-data-resource__button">
+          <ListPageCreateLink to={createLink}>
+            {t("Create Data resource")}
+          </ListPageCreateLink>
+        </div>
+      </div>
+      <ListPageBody>
+        <ResourceTable
+          data={filteredData}
+          unfilteredData={data}
+          loaded={loaded}
+          loadError={loadError}
+          rowData={{
+            launchModal,
+            kebabActions,
+            bucketPolicyMap
+          }}
+        />
+      </ListPageBody>
+    </>
+  );
+};
+
+const GenericListPage: React.FC<GenericListPageProps> = () => {
+  return (
+    <>
+      <div className="center-component">
+        <>{"Data Federation ~ Sample List View"}</>
+      </div>
+    </>
+  );
+};
+export const BackingStoreListPage: React.FC = () => (
+  <GenericListPage resourceModel={NooBaaBackingStoreModel} />
+);
+
+export const BucketClassListPage: React.FC = () => (
+  <GenericListPage resourceModel={NooBaaBucketClassModel} />
+);
```

Comment:
are these lines required?

Reply:
these are added and will be taken care by @SanjalKatiyar

Reply:
I'll suggest to add a comment  in that case so we don't miss it

Reply:
moved data resources to a separate file, later when @SanjalKatiyar  will work he can have a look into generics stuff and bring them onto the generic list page file.